runtime/src/iree/hal/allocator_heap.c

```diff
@@ -118,11 +118,16 @@ iree_hal_heap_allocator_query_compatibility(
   // much easier to find issues of buffer definition with local devices that
   // will cause issues when used with real devices.
   if (iree_all_bits_set(params->type, IREE_HAL_MEMORY_TYPE_DEVICE_VISIBLE)) {
-    if (iree_all_bits_set(params->usage, IREE_HAL_BUFFER_USAGE_TRANSFER)) {
```

Comment:
ah this should be `if (iree_any_bits_set(params->usage, IREE_HAL_BUFFER_USAGE_TRANSFER)) {`
(and same for the others) - then no need for the expanded comparisons and in the future if we add more transfer/storage/etc types we don't need to update all these locations.